src/pages/Inventory.page.js

```diff
@@ -9,7 +10,102 @@ export class InventoryPage extends BaseSwagLabPage {
+
+    async getItemName(item) {
+        return item.locator(this.itemName).textContent();
+    }
+
+    async getItemDescription(item) {
+        return item.locator(this.itemDescription).textContent();
+    }
+
+    async getItemPrice(item) {
+        return item.locator(this.itemPrice).textContent();
+    }
+
     async addItemToCartById(id) {
         await this.addItemToCartButton.nth(id).click();
     }
+
+    async selectOption(optionValue) {
+        return this.productSortContainer.selectOption(optionValue);
+    }
+
+    async getItemNames() {
+        return this.itemName.evaluateAll(items =>
+            items.map(item => item.textContent.trim()));
+    }
+
+    async getItemPrices() {
+        return this.itemPrice.evaluateAll(items =>
+            items.map(item => parseFloat(item.textContent.replace('$', '')))
+        );
+    }
+
+    async expectSortedProducts(sortBy, productNames, productPrices) {
+        switch (sortBy) {
+          case 'Name (A to Z)':
+            const sortedNamesAZ = await this.getItemNames();
+            const expectedNamesAZ = productNames.sort();
+            expect(sortedNamesAZ).toEqual(expectedNamesAZ);
+            break;
+          case 'Name (Z to A)':
+            const sortedNamesZA = await this.getItemNames();
+            const expectedNamesZA = productNames.sort().reverse();
+            expect(sortedNamesZA).toEqual(expectedNamesZA);
+            break;
+          case 'Price (low to high)':
+            const sortedPricesLowHigh = await this.getItemPrices();
+            const expectedPricesLowHigh = productPrices.sort((a, b) => a - b);
+            expect(sortedPricesLowHigh).toEqual(expectedPricesLowHigh);
+            break;
+          case 'Price (high to low)':
+            const sortedPricesHighLow = await this.getItemPrices();
+            const expectedPricesHighLow = productPrices.sort((a, b) => b - a);
+            expect(sortedPricesHighLow).toEqual(expectedPricesHighLow);
+            break;
+          default:
+            throw new Error('Sort option is not correct');
+        }
+      }
+
+      async pickRandomItems(numberOfProducts, productCount) {
+        const randomNumbers = new Set();
+        while (randomNumbers.size < Math.min(numberOfProducts, productCount)) {
+            randomNumbers.add(Math.floor(Math.random() * productCount));
+        }
+        return Array.from(randomNumbers);
+    }
+
+      async addRandomProductsToCart(productCount) {
+        const uniqueRandomNumbers = await this.pickRandomItems(3, productCount);
+        const selectedProducts = [];
+
+        for (const index of uniqueRandomNumbers) {
+            const product = await this.inventoryItems.nth(index);
+
+            const name = await this.getItemName(product);
+            const description = await this.getItemDescription(product);
+            const price = await this.getItemPrice(product);
+
+            console.log(`Selected product name: ${name}`);
```

Comment:
Do we need this console.log? There is a better way to store these data in reports. For example 

```
await test.info().attach('Added random product', { body: JSON.stringify(products, null, 4) });
```

in this way you can see data you have chosen in the report

Reply:
This approach can be used for the other console logs too